### PR TITLE
Remove lodash

### PR DESCRIPTION
**Pull request by misteroneill**
## Description
This removes lodash-compat as a dependency (though, lodash is preserved as a dev dep). _This reduces the minified+gzipped size of Video.js by 4KB._

## Specific Changes proposed
- Remove lodash-compat as a dependency.
- Add lodash as a dev dependency.
- Reimplement a pared down throttle function in our `fn` utility.
- Reimplement `_.merge` behavior in `mergeOptions`.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by misteroneill** [comment]
Note: this includes commits from #3757

File: src/js/utils/obj.js [hunk 1]
```diff
@@ -30,3 +31,55 @@ export function reduce(object, fn, initial = 0) {
...
 }
+
+/**
+ * Object.assign-style object shallow merge/extend.
+ *
+ * @param  {Object} target
+ * @param  {Object} ...sources
+ * @return {Object}
+ */
+export function assign(target, ...sources) {
```
Comment (mmcc) [inline src/js/utils/obj.js]: FWIW, I use a similar helper extensively at work (Redux app) and we went ahead and defaulted to setting the target to an empty object so the result is always a shallow copy. Not sure if that makes sense here, though.

**Comment by boushley** [comment]
Have we checked the size difference of using our own vs using the individual lodash packages like: https://www.npmjs.com/package/lodash.throttle and

**Comment by boushley** [comment]
Dang, fat fingered the submit button instead of paste. That's what I get for looking at issues on my phone...

https://www.npmjs.com/package/lodash.assign

**Comment by misteroneill** [comment]
@boushley It's a good thought. I did an experiment a few weeks ago with those packages but they have internal dependencies (internal lodash functions, etc) that add a lot more weight than these implementations. lodash.assign for example weighs in at 1.6KB minified and gzipped, which would negate the file size savings.

File: src/js/utils/obj.js (hunk 1 above)
Comment (misteroneill) [inline src/js/utils/obj.js]: How does that work if the target is the first argument and the sources are the rest?

**Comment by boushley** [comment]
@misteroneill glad to hear you already investigated it. I suspected there was something like that :)

Given that's been investigated I think these are great alternatives that save quite a bit of bulk :shipit:

File: src/js/utils/obj.js (hunk 1 above)
Comment (gkatsev) [inline src/js/utils/obj.js]: I think it basically ends up being `assign({}, target, source1, source2)`.

Comment (mmcc) [inline src/js/utils/obj.js]: ^^ this.

Comment (gkatsev) [inline src/js/utils/obj.js]: However, there's one nice usecase of not forcing target to be an empty object:
```js
Object.assign(el.style, {
  'color': 'blue',
  'background': 'red'
});
```
Or similar. But yeah, *most* times, it probably does want to just be an empty object.

Comment (misteroneill) [inline src/js/utils/obj.js]: Could have another method - `assignNew` or something. That's a bad name, but you get my drift. Sometimes it's convenient to mutate an object.

**Pull request merged**